packages/atomic/src/components/product-listing/atomic-product-listing-result-template/atomic-product-listing-result-template.tsx

```diff
@@ -0,0 +1,69 @@
+import {ResultTemplate, ResultTemplateCondition} from '@coveo/headless';
+import {Component, Element, Prop, Method, State} from '@stencil/core';
+import {MapProp} from '../../../utils/props-utils';
+import {
+  ResultTemplateCommon,
+  makeMatchConditions,
+} from '../../common/result-templates/result-template-common';
+
+/**
+ * The `atomic-product-listing-template` component determines the format of the query results, depending on the conditions that are defined for each template. A `template` element must be the child of an `atomic-product-listing-template`, and either an `atomic-result-list` or `atomic-folded-result-list` must be the parent of each `atomic-result-template`.
+ *
+ * Note: Any `<script>` tags defined inside of a `<template>` element will not be executed when results are being rendered.
+ * @MapProp name: mustMatch;attr: must-match;docs: The field and values that define which result items the condition must be applied to. For example, a template with the following attribute only applies to result items whose `filetype` is `lithiummessage` or `YouTubePlaylist`: `must-match-filetype="lithiummessage,YouTubePlaylist"`;type: Record<string, string[]> ;default: {}
+ * @MapProp name: mustNotMatch;attr: must-not-match;docs: The field and values that define which result items the condition must not be applied to. For example, a template with the following attribute only applies to result items whose `filetype` is not `lithiummessage`: `must-not-match-filetype="lithiummessage";type: Record<string, string[]> ;default: {}
+ */
+@Component({
```

Comment:
For all components that you are creating as part of this whole task about product listing, I would suggest adding a JSDOC comment `@internal`. 

You should see this pattern used for `insight` use case/components.

This will mark the component still under development/not yet ready for production, and will allow you to merge in master continuously (and not block any release) while the work is still ongoing.

You could do breaking changes/completely rework all components and it would be fine.


The components will be "available" to use if you know they exists (ie: you could install Atomic from NPM/production CDN and use the components for a demo on your side or something similar), but they won't appear in our doc, they won't generate code for Angular or React packages, etc.

Then, as a work item, when everything is good to go in production and is stable, you could go and remove all the `@internal` tags in one PR.

Reply:
Sounds good, thanks